lib/instance.js

```diff
@@ -127,7 +127,7 @@ Instance.prototype.where = function() {
   var where;
 
   where = this.Model.primaryKeyAttributes.reduce(function (result, attribute) {
-    result[attribute] = this.get(attribute, {raw: true});
+    result[this.Model.rawAttributes[attribute].field] = this.get(attribute, {raw: true});
```

Comment:
I'm not too happy about that, it will confuse users who use `.where()` expecting attributes and not fields like we normally use.

You can use `Utils.mapOptionFieldNames` (https://github.com/sequelize/sequelize/blob/master/lib/utils.js#L80) instead in the destroy code.